### DISCUSSION TITLE
Vitest does not exit when using inline tests.

### DISCUSSION THREAD

**Discussion by brielov** [vitest-dev/vitest#1217] [Q&A]
Hi there, I'm trying to migrate a package of mine called [typed](https://github.com/brielov/typed) to vite and vitest and when running tests one file at a time, it runs well, but when running everything it hangs.

```ts
// vite.config.ts

/// <reference types="vitest" />

import dts from "vite-plugin-dts";
import { defineConfig } from "vite";
import { resolve } from "path";

export default defineConfig({
  plugins: [dts()],
  build: {
    target: "es2020",
    lib: {
      entry: resolve(__dirname, "src/index.ts"),
      name: "typed",
      formats: ["es", "umd"],
      fileName: "index",
    },
  },
  define: {
    "import.meta.vitest": false,
  },
  test: {
    includeSource: ["src/**/*.ts"],
    watch: false,
  },
});
```

And here is the `npm run test` output:

```
 √ src/error.ts (2)
 √ src/structs/boolean.ts (2)
 √ src/structs/date.ts (7)
 √ src/structs/any.ts (9)
 √ src/structs/defaulted.ts (3)
 √ src/structs/array.ts (3)
 √ src/structs/enums.ts (2)
 √ src/structs/literal.ts (3)
 √ src/structs/number.ts (4)
 √ src/structs/intersection.ts (2)
 √ src/structs/object.ts (3)
 √ src/structs/record.ts (5)
 √ src/structs/optional.ts (3)
 √ src/structs/nullable.ts (3)
 √ src/structs/tuple.ts (4)
 √ src/structs/string.ts (3)
 √ src/structs/unknown.ts (9)
 √ src/structs/union.ts (4)
```

But it never exits. Causing ci to fail eventually. 

I'm using inline tests on each file.

**Comment by brielov** [comment]
I ran each file and there's one missing called `src/util.ts`. It hangs on that file when using `await import('..')`. This does not happen on any other file.

**Reply by sheremet-va** [reply]
does “../index” imports utils? it might be importing itself in the process and just hangs

**Reply by brielov** [reply]
@sheremet-va I don't think so. https://github.com/brielov/typed/blob/feat/vite/src/index.ts

**Reply by sheremet-va** [reply]
what do you mean? it clearly imports itself :)

inside utils you import index, which also imports utils. so it’s stack in a loop. what wierd is that vitest didn’t threw an error

**Reply by brielov** [reply]
Damn, I was importing test-util from util and util from test-util 🙃. I ended up moving from inline style testing to traditional because it broke vite build. Thanks for the help!
